plugins/cpp_metrics/parser/src/cppmetricsparser.cpp

```diff
@@ -165,6 +168,170 @@ void CppMetricsParser::functionBumpyRoad()
   });
 }
 
+template <typename T>
+void CppMetricsParser::checkTypes(
+  const std::string path,
+  const std::unordered_set<std::uint64_t>& typesFound,
+  const std::unordered_map<std::uint64_t,std::string>& typeDefinitionPaths,
+  std::unordered_map<std::string,std::unordered_set<std::uint64_t>>& relationsFoundInFile,
+  int& relationsInModule
+)
+{
+  for (const T& item
+    : _ctx.db->query<T>(
+      odb::query<T>::query_columns::File::path.like(path + "%")
+    ))
+    {
+      if (
+        typesFound.find(item.typeHash) != typesFound.end() &&
+        //check if return type is defined within the current module
+        typeDefinitionPaths.at(item.typeHash) != item.filePath &&
+        //check for self relation
+        (relationsFoundInFile.find(item.filePath) == relationsFoundInFile.end() ||
+        //check if any reletaions were found in the current file
+        relationsFoundInFile[item.filePath].find(item.typeHash) == relationsFoundInFile[item.filePath].end())
+        //check if this relation is already found
+      )
+      {
+        ++relationsInModule;
+        auto it = relationsFoundInFile.find(item.filePath);
+        if (it != relationsFoundInFile.end())
+        {
+          it->second.insert(item.typeHash);
+        }
+        else
+        {
+          relationsFoundInFile.insert(std::make_pair(item.filePath,std::unordered_set<std::uint64_t>{item.typeHash}));
+        }
+      }
+    }
+}
+
+void CppMetricsParser::relationalCohesion()
+{
+  std::unordered_set<std::string> filepaths;
+  std::unordered_map<std::uint64_t,std::string> typeDefinitionPaths;
+  std::unordered_set<std::uint64_t> typesFound;
+  std::unordered_map<std::string,std::uint64_t> moduleIds;
+
+  util::OdbTransaction {_ctx.db} ([&, this]
+  {
+
+    std::ifstream file(_modulesPath);
+
+    //read the specified module list if given
```

Comment:
Coding convention for comments is the following:
```
// Read the specified module list if given
```
(Note the space and the capitalization.)

Please apply this to all your in-code comments.